src/server/acl/user_registry.h

```diff
@@ -48,8 +53,21 @@ class UserRegistry {
   // Used by Auth
   bool AuthUser(std::string_view username, std::string_view password) const;
 
+  // Helper class for accessing the registry with a ReadLock outside the scope of UserRegistry
+  class RegistryViewWithLock {
+   public:
+    RegistryViewWithLock(std::shared_lock<util::SharedMutex> mu, const RegistryType& registry);
+    const RegistryType& registry;
+
+   private:
+    std::shared_lock<util::SharedMutex> registry_mu_;
+  };
+
+  // Helper function used for printing users via ACL LIST
+  RegistryViewWithLock GetRegistryWithLock() const;
```

Comment:
This is actually a nice pattern 👍🏻 , in many places we have custom accessors that pre-transform the data, but its not always the best solution

Reply:
Yes and no. I like the pattern, but for this specific case I am 50-50 because originally, I wanted to restrict the `locks` only within the member functions of the `UserRegistry` -- that's why I design it with the `UpdateRequest` in mind, such that all the updates are done internally by the member function in one step.

This is an exception, since printing and formatting is not a logical part of `UserRegistry` but of the `ACL LIST` command. That being said, this function is an exception (and kinda orthogonal) to the design philosophy of this class. 

All in all, I could live with this exception, since this is only used by only one command. If this changes, I would look for alternatives.